### ISSUE TITLE
Support token based auth

### ISSUE THREAD

**Issue by betatim** [osfclient/osfclient#101]
We should support token based auth in addition to password based. See for example #99

https://developer.osf.io/#Introduction_authentication shows which header needs to change.

**Comment by DimitriPapadopoulos** [comment]
As far as I can understand the _OSF.__init__()_ and _OSF.login()_ methods do accept a token parameter for that purpose, but it is currently unused.

I access my OSF account using ORCID, I don't have a password. Is there any way to upload file to my OSF project?

**Comment by wikey** [comment]
As long as you *can* log in to OSF, you can generate a personal access token through this link: https://osf.io/settings/tokens/

That token can then be fed to the API instead of a password.

**Comment by DimitriPapadopoulos** [comment]
I do see that a token can be fed to the API, to [_OSF.\_\_init\_\_()_](https://github.com/osfclient/osfclient/blob/44b9a87/osfclient/api.py#L13-L16) and [_OSF.login()_](https://github.com/osfclient/osfclient/blob/44b9a87/osfclient/api.py#L18-L20), but then _token_ is **not** used as far as I can see. That would be the whole purpose of pull request #163. I'll give it a try though, but I doubt it works.

**Comment by DimitriPapadopoulos** [comment]
Tried this:
```python
def main():
    osf = OSF()
    osf.login(USERNAME, token=TOKEN)
    osf.project('8ch5y')


if __name__ == '__main__':
    main()
```
The error message is:
```python
Traceback (most recent call last):
  File "./osf2.py", line 15, in <module>
    main()
  File "./osf2.py", line 11, in main
    osf.project('8ch5y')
  File "/home/dimitri/.local/lib/python3.6/site-packages/osfclient/api.py", line 24, in project
    return Project(self._json(self._get(url), 200), self.session)
  File "/home/dimitri/.local/lib/python3.6/site-packages/osfclient/models/core.py", line 23, in _get
    return self.session.get(url, *args, **kwargs)
  File "/home/dimitri/.local/lib/python3.6/site-packages/osfclient/models/session.py", line 45, in get
    raise UnauthorizedException()
osfclient.exceptions.UnauthorizedException
```

**Comment by devinberg** [comment]
Would the use of a token permit _osfclient_ to work with accounts that have 2FA enabled also?

**Comment by felliott** [comment]
This should be fixed in the next release, which I am preparing now.

@devinberg, yes!  You create a token via regular login, which is protected by 2FA.  But using the token after that does not require 2FA.

**Comment by felliott** [comment]
This has been merged into `master` and released in `osfclient==0.0.5`. Thank you, all!
